axlearn/common/trainer.py

```diff
@@ -560,6 +560,11 @@ def run(
                 output = None
                 stop_trace_step = None
 
+                # skip previous batches
```

Comment:
Does Axlearn not already do this?

Reply:
axlearn provide a flag `save_input_iterator` which was supposed to save the input iterator as part of the checkpoint to achieve the same functionality (otherwise, training always restart from the beginning of the data). but it is running into exception as commented in their repo [here](https://github.com/apple/axlearn/blob/main/axlearn/common/trainer.py#L153). I encountered the same exception with this flag, so added the logic to skip batches.